app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -534,6 +536,10 @@ private void configureRightCard() {
     private void configureNavigationButtons() {
         // Navigation next / previous for each image as we're collecting title + description
         next.setOnClickListener(v -> {
+            if (DeviceInfoUtil.getConnectionType(this) == ConnectionType.NO_INTERNET) {
```

Comment:
Is the check needed in the first step of the upload? Ideally, it can be moved to the last step.

Reply:
The app is crashing during the first step.